app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -1063,18 +1061,13 @@ protected static function isCaptureRequest($newTransactionStatus, $prevTransacti
     public function handleVoidTransactionUpdate(Mage_Payment_Model_Info $payment){
 
         $authTransaction = $payment->getAuthorizationTransaction();
+        /** @var Mage_Sales_Model_Order $order */
         $order = $payment->getOrder();
         $amount =  $order->getBaseGrandTotal() - $order->getBaseTotalPaid() ;
 
         if (!$authTransaction || $authTransaction->canVoidAuthorizationCompletely()) {
             // True void
-            $reference = $payment->getAdditionalInformation('bolt_reference');
-            $payment->setParentTransactionId($reference);
-            $payment->setTransactionId(sprintf("%s-void", $reference));
-            $payment->setShouldCloseParentTransaction(true);
-            $transaction = $payment->addTransaction(Mage_Sales_Model_Order_Payment_Transaction::TYPE_VOID, null, true);
-            $message = $this->getVoidMessage($payment, $transaction, $amount);
-            $order->setState(Mage_Sales_Model_Order::STATE_CANCELED, true, $message);
```

Comment:
Does `$order->cancel();` include the `setState` code? If it does, then it should be removed on lines 499-500 above as well.

Reply:
@matt-thomason Nice 👀 and nice thought.  For this case,

https://github.com/BoltApp/bolt-magento1/blob/55d4e4cfc878612f4bd9822f6a967d3858182609/app/code/community/Bolt/Boltpay/Model/Payment.php#L498-L500

we are not so much interested in setting the state, but rather appending a message to the order history that the order was permanently rejected by Bolt.  I kept it as is for the side effect, but, we can make it more semantically and possibly slightly more performance efficient.  

Hold one moment.